plugins/inputs/lustre2/lustre2.go

```diff
@@ -440,6 +464,98 @@ func (l *Lustre2) GetLustreProcStats(fileglob string, wantedFields []*mapping) e
 	return nil
 }
 
+func (l *Lustre2) getLustreProcBrwStats(fileglob string, wantedFields []*mapping) error {
+	files, err := filepath.Glob(fileglob)
+	if err != nil {
+		return fmt.Errorf("failed to find files matching glob %s: %w", fileglob, err)
+	}
+
+	for _, file := range files {
+		// Turn /proc/fs/lustre/obdfilter/<ost_name>/stats and similar into just the object store target name
+		// This assumes that the target name is always second to last, which is true in Lustre 2.1->2.12
+		path := strings.Split(file, "/")
+		if len(path) < 2 {
+			continue
+		}
+		name := path[len(path)-2]
+
+		wholeFile, err := os.ReadFile(file)
+		if err != nil {
+			return fmt.Errorf("failed to read file %s: %w", file, err)
+		}
+		lines := strings.Split(string(wholeFile), "\n")
+
+		var headerName string
+		for _, line := range lines {
+			// There are four types of lines in a brw_stats file:
+			// 1. Header lines - contain the category of metric (e.g. disk I/Os in flight, disk I/O time)
+			// 2. Bucket lines - follow headers, contain the bucket value (e.g. 4K, 1M) and metric values
+			// 3. Empty lines - these will simply be filtered out
+			// 4. snapshot_time line - this will be filtered out, as it "looks" like a bucket line
+			if len(line) < 1 {
+				continue
+			}
+			parts := strings.Fields(line)
+
+			// This is a header line
+			// Set report name for use by the buckets that follow
+			if !strings.Contains(parts[0], ":") {
+				nameParts := strings.Split(line, "  ")
+				headerName = nameParts[0]
+				continue
+			}
+
+			// snapshot_time should be discarded
+			if strings.Contains(parts[0], "snapshot_time") {
+				continue
+			}
+
+			// This is a bucket for a given header
+			for _, wanted := range wantedFields {
+				if headerName != wanted.inProc {
+					continue
+				}
```

Comment:
Wouldn't it be easier to make `wantedFields` a map and look-up the `headerName`?

Reply:
I agree, I don't like the way this code is laid out currently with the field mappings and I plan to refactor it in a follow-up PR. For now I just followed how the other mappings were done, but I can refactor this here instead if you'd prefer